relm4/src/factory/sync/collections/hashmap.rs

```diff
@@ -0,0 +1,239 @@
+use crate::Sender;
+
+use crate::factory::sync::builder::FactoryBuilder;
+use crate::factory::sync::handle::FactoryHandle;
+use crate::factory::{CloneableFactoryComponent, FactoryComponent, FactoryView};
+
+use std::collections::HashMap;
+use std::hash::Hash;
+use std::iter::FusedIterator;
+use std::ops;
+
+#[derive(Debug)]
+#[must_use]
+pub struct FactoryElementGuard<'a, C>
+where
+    C: FactoryComponent,
+{
+    inner: &'a mut FactoryHandle<C>,
+}
+
+impl<'a, C> ops::Deref for FactoryElementGuard<'a, C>
+where
+    C: FactoryComponent,
+{
+    type Target = C;
+
+    fn deref(&self) -> &Self::Target {
+        self.inner.data.get()
+    }
+}
+
+impl<'a, C> ops::DerefMut for FactoryElementGuard<'a, C>
+where
+    C: FactoryComponent,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.inner.data.get_mut()
+    }
+}
+
+impl<'a, C> Drop for FactoryElementGuard<'a, C>
+where
+    C: FactoryComponent,
+{
+    fn drop(&mut self) {
+        self.inner.notifier.send(()).unwrap()
+    }
+}
+
+/// A container similar to [`HashMap`] that can be used to store
+/// values of type [`FactoryComponent`].
+#[derive(Debug)]
+pub struct FactoryHashMap<K, C: FactoryComponent> {
+    widget: C::ParentWidget,
+    parent_sender: Sender<C::ParentInput>,
+    inner: HashMap<K, FactoryHandle<C>>,
```

Comment:
I am being a bit demanding, but it would be really great if different hash types could be used.

Reply:
Can you elaborate what you mean with different hash types? I'm not sure if I understand what you mean.

Reply:
I think you meant different hashers, right? Anyway, I've just added support for that.